### ISSUE TITLE
openMM snapshot.masses fails for snapshots with TopologyEngine and virtual sites

### ISSUE THREAD

**Issue by hejung** [openpathsampling/openpathsampling#1084]
Stumbled across this running with TIP4P and random velocities :)
To reproduce load any pdb/trajectory with TIP4P and access the `masses` or `masses_per_mole` attributes of the snapshots. I attached a pdb with TIP4P waters (as txt because github doesnt like pdb). [LiCl_solv_tip4pew.txt](https://github.com/openpathsampling/openpathsampling/files/7426721/LiCl_solv_tip4pew.txt)
```python
import openpathsampling.engines.openmm as peng_omm
template = peng_omm.snapshot_from_pdb('LiCl_solv_tip4pew.pdb')
template.masses  # this will fail
```
The issue here is that in [masses.py](https://github.com/openpathsampling/openpathsampling/blob/ea605e8b8305029e3e52b83615405a8f6aed3377/openpathsampling/engines/openmm/features/masses.py) if the snapshot engine has no attached simulation we do the following to get the masses:
```python
ops_topology = snapshot.topology
topology = ops_topology.mdtraj.to_openmm()
masses_per_mole = [a.element.mass for a in topology.atoms()]
```
Which fails for virtual sites because they have `None` as element in the openmm-topology.

Opening this issue because I am not sure what the best way to address this properly would be and I already found a quick and dirty fix. To work around it temporarily one can (re)set the engine to a `OpenMMEngine` with an attached simulation object to take the different code path in  [masses.py](https://github.com/openpathsampling/openpathsampling/blob/ea605e8b8305029e3e52b83615405a8f6aed3377/openpathsampling/engines/openmm/features/masses.py).

Concerning possible solutions: In mdtraj the virtual sites do have an element ("VS") with a mass of zero, the `None` element is only in the openmm topology. I guess we are converting to openmm topology before getting the masses to make sure we have exactly the mass the engine uses, so I am not sure if using the mdtraj elements and masses (i.e. without the conversion) would be an option. And I am not sure if there are other situations in which an openmm topology returns `None` for the atom element, so I am not sure if something like `mass=0 if a.element is None` would be an option.
[Edit: typos]

**Comment by sroet** [comment]
@hejung, can this issue be closed after the merging of #1085 ?

**Comment by hejung** [comment]
@sroet yes! sorry for not doing that myself.
